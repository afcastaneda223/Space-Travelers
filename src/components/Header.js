import React from 'react';
import Navbar from './Navbar';

function Header() {
  return (
    <header className="sticky-top">
      <Navbar />
    </header>
  );
}

export default Header;
