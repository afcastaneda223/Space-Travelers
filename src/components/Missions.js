import React from 'react';

function Missions() {
  return (
    <div>
      Mission here!
    </div>
  );
}

export default Missions;
