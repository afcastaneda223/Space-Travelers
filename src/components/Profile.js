import React from 'react';
import ProfileMissions from './ProfileMissions';
import ProfileRockets from './ProfileRockets';

function Profile() {
  return (
    <div className="container">
      <div className="row row-cols-lg-2 row-cols-1">
        <div className="col">
          <h3>My Missions</h3>
          <ProfileMissions />
        </div>
        <div className="col">
          <h3>My Rockets</h3>
          <ProfileRockets />
        </div>
      </div>
    </div>
  );
}

export default Profile;
