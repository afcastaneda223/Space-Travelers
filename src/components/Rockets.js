import React from 'react';
import RocketList from './RocketList';

function Rockets() {
  return (
    <div className="container">
      <RocketList />
    </div>
  );
}

export default Rockets;
