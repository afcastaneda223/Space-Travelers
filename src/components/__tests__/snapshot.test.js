import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import Rockets from '../Rockets';
import Missions from '../Missions';
import Profile from '../Profile';
import Header from '../Header';
import store from '../../redux/configureStore';

describe('Render Rockets page', () => {
  it('Renders the Rockets page', () => {
    const rockets = TestRenderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(rockets).toMatchSnapshot();
  });
});

describe('Render Missions page', () => {
  it('Renders the Missions', () => {
    const missions = TestRenderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});

describe('Render Profile page', () => {
  it('Renders the profile', () => {
    const profile = TestRenderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(profile).toMatchSnapshot();
  });
});

describe('Render Navbar', () => {
  it('Renders the navbar', () => {
    const header = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(header).toMatchSnapshot();
  });
});
