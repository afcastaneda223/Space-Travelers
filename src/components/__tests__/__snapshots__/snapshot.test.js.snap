// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Missions page Renders the Missions 1`] = `
<div
  className="container border border-3"
>
  <div
    className="row row-cols-1 row-cols-md-12"
  >
    <div
      className="col-md-2 border border-1"
    >
      <strong>
        Mission
      </strong>
    </div>
    <div
      className="col-md-6 border border-1"
    >
      <strong>
        Description
      </strong>
    </div>
    <div
      className="col-md-2 border border-1"
    >
      <strong>
        Status
      </strong>
    </div>
    <div
      className="col-md-2 border border-1"
    />
  </div>
</div>
`;

exports[`Render Navbar Renders the navbar 1`] = `
<header
  className="sticky-top"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light mb-5 sticky-top"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Logo"
          src="planet.png"
        />
        <button
          className="btn btn-light"
          type="button"
        >
          <h3>
            Space Travelers' Hub
          </h3>
        </button>
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse justify-content-end"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav text-end mb-2 mb-lg-0 pe-5"
        >
          <li
            className="m-2"
          >
            <a
              aria-current="page"
              className="link-dark text-decoration-none fw-bold text-decoration-underline"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Rockets
            </a>
          </li>
          <li
            className="m-2"
          >
            <a
              aria-current={null}
              className="link-dark text-decoration-none"
              href="/missions"
              onClick={[Function]}
            >
              Missions
            </a>
          </li>
          <li
            className="m-2"
          >
            <a
              aria-current={null}
              className="link-dark text-decoration-none"
              href="/profile"
              onClick={[Function]}
            >
              Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Render Profile page Renders the profile 1`] = `
<div
  className="container"
>
  <div
    className="row row-cols-lg-2 row-cols-1"
  >
    <div
      className="col"
    >
      <h3>
        My Missions
      </h3>
      <div>
        <ul
          className="list-group"
        />
      </div>
    </div>
    <div
      className="col"
    >
      <h3>
        My Rockets
      </h3>
      <div>
        <ul
          className="list-group"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Rockets page Renders the Rockets page 1`] = `
<div
  className="container"
>
  <div />
</div>
`;
